### ISSUE TITLE
Cannot use StatsigProvider

### ISSUE THREAD

**Issue by juanstiza** [statsig-io/react-sdk#3]
Hi!

We are trying to use your app, but we are getting an error:

```
Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

Our code looks like this:

```typescript
import { StatsigProvider } from 'statsig-react-native';
import { Text } from 'react-native';

export default function App() {
  return (
    <StatsigProvider
      sdkKey="client-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
      user={{ userID: 'test' }}
      waitForInitialization={true}
    >
      <Text>Hello</Text>
    </StatsigProvider>
  );
}
```

We use an ejected Expo project. Our dependencies:

```
"react-native": "~0.63.4",
"@react-native-async-storage/async-storage": "^1.15.6",
"react-native-device-info": "^8.3.0",
"react-native-get-random-values": "^1.7.0",
"statsig-react-native": "^3.0.0",
```

**Comment by tore-statsig** [comment]
Thanks for reporting @juanstiza - I will start investigating this.  In the mean time, you may want to check out the expo sdk (though I'm not sure what the implications of ejecting an expo project are in terms of which SDK would be a better fit)

https://github.com/statsig-io/react-native-expo

**Comment by tore-statsig** [comment]
Alright, updating with some of my initial findings.

So far I have been unable to reproduce the issue, but the new project I spun up to check this was using react-native version 0.65.0 (that was the only difference)

I *think* this is going to be related to the the hooks error due to multiple react versions or mismatched versions.  The statsig-react sdk correctly lists react as a peer dependency, but it is also listed as a dependency here:

https://github.com/statsig-io/react-sdk/blob/main/package.json#L32

This is my best lead so far.  While I try to get a repro working so I can test that theory and then bump the statsig-react and statsig-react-native SDK versions, you might try some of the workarounds listed here:

https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react

or on this github issue:

https://github.com/facebook/react/issues/13991

**Comment by tore-statsig** [comment]
I was still unable to reproduce the issue this afternoon.

Given my initial guess, I updated the react sdk to 0.5.1, and the statsig-react-native sdk to 3.0.1 to remove react as a dependency of the sdk (and only mark it as a peer dependency)

When you get a chance, please try installing via:

npm install statsig-react-native@3.0.1

yarn upgrade statsig-react-native@3.0.1

and comment back here if you are facing the same problem

**Comment by tore-statsig** [comment]
Looks like that did the trick, and this has been fixed in v3.0.2
